cross-dev/build_docker_image.rs

```diff
@@ -0,0 +1,158 @@
+use std::{path::Path, process::Command};
+
+use clap::Args;
+use cross::CommandExt;
+
+#[derive(Args, Debug)]
+pub struct BuildDockerImage {
+    #[clap(long, hide = true, env = "GITHUB_REF_TYPE")]
+    ref_type: Option<String>,
+    #[clap(long, hide = true, env = "GITHUB_REF_NAME")]
+    ref_name: Option<String>,
+    /// Newline separated labels
+    #[clap(long, env = "LABELS")]
+    labels: Option<String>,
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    verbose: bool,
+    #[clap(long)]
+    force: bool,
+    #[clap(short, long)]
+    push: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+
+    /// Targets to build for
+    #[clap()]
+    targets: Vec<String>,
+}
+
+pub fn build_docker_image(
+    BuildDockerImage {
+        mut targets,
+        verbose,
+        force,
+        push,
+        ref_type,
+        ref_name,
+        labels,
+        ..
+    }: BuildDockerImage,
+    engine: &Path,
+) -> cross::Result<()> {
+    let metadata = cross::cargo_metadata_with_args(
+        Some(Path::new(env!("CARGO_MANIFEST_DIR"))),
+        None,
+        verbose,
+    )?
+    .ok_or_else(|| eyre::eyre!("could not find cross workspace and its current version"))?;
+    let version = metadata
```

Comment:
This isn't necessarily correct: if a build script is used or another package is compiled earlier, then this won't be the package desired.

If you have a sample package with a `cc` dependency in a build script, which we might add eventually, then the output of the following will be something as follows:

**Code**

```rust
    let metadata = cargo_metadata_with_args(
        Some(Path::new(env!("CARGO_MANIFEST_DIR"))),
    ).unwrap()
    .ok_or_else(|| eyre::eyre!("could not find cross workspace and its current version")).unwrap();
    eprintln!("metadata.packages={:?}", metadata.packages);
```

**Output**

```text
[Package { id: "cc 1.0.73 (registry+https://github.com/rust-lang/crates.io-index)", manifest_path: "/home/ahuszagh/.cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.73/Cargo.toml", source: Some("registry+https://github.com/rust-lang/crates.io-index"), version: "1.0.73" }
```

```

Reply:
I can't replicate this scenario, we're running `cargo metadata --no-deps` so the extra package will never show up

Reply:
Oh that's weird, it wasn't working on mine. Hmmm maybe I thought `--no-deps` was getting added when it wasn't.